### ISSUE TITLE
Reconsider the usage of "dialogs" for non-modal sidebar panels

### ISSUE THREAD

**Issue by acelaya** [hypothesis/client#6307]
We have discussed that perhaps sidebar panels should not follow a non-modal dialog pattern, as it can cause confusions.

Some alternative patterns we could consider are:

* Named regions: https://whatsock.com/Templates/Popups/index.htm
* Disclosure: https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/

There's more context in our internal slack, starting on this message: https://hypothes-is.slack.com/archives/C1M8NH76X/p1712049084215899?thread_ts=1711380668.404809&cid=C1M8NH76X

**Comment by acelaya** [comment]
> as it can cause confusions

I wish I had documented what were those confusions 😅 

I suppose I can I only guess now, hehe.

**Comment by acelaya** [comment]
We have refactored the sidebar panels to no longer have `dialog` role, but `region` role instead.

When opened, they are automatically focused, and screen readers announce them as landmarks with a name.
